mono/mini/mini-amd64.c

```diff
@@ -1443,7 +1443,7 @@ mono_arch_compute_omit_fp (MonoCompile *cfg)
 
 	if (!cfg->arch.cinfo)
 		cfg->arch.cinfo = get_call_info (cfg->mempool, sig);
-	cinfo = (CallInfo *)cfg->arch.cinfo;
+	cinfo = cfg->arch.cinfo;
```

Comment:
If we forward declare CallInfo, then why are these casts needed ?

Reply:
The intent (did I mess up?) is to remove casts, not add them.
And then to make it more valid C++ in the process -- but better C either way imho.

Reply:
Misread it.

Reply:
Oh -- perhaps, you mean -- we could cut out local variables in the process? Yes we probably could.